src/main/java/seedu/address/logic/commands/NearbyCommand.java

```diff
@@ -4,75 +4,153 @@
 import static seedu.address.logic.commands.NearbyCommandUtil.getGeneralLocation;
 import static seedu.address.logic.commands.NearbyCommandUtil.isValidPostalSector;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 import java.util.Optional;
 import java.util.function.Predicate;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 import seedu.address.commons.core.index.Index;
 import seedu.address.logic.commands.exceptions.CommandException;
 import seedu.address.model.Model;
 import seedu.address.model.order.Order;
 
 /**
- * Used to identify orders in the order book that belong to a given postal sector.
+ * Used to identify orders in the order book that belong to a given postal sector or area.
  */
 public class NearbyCommand extends Command {
     public static final String COMMAND_WORD = "nearby";
     public static final String NEWLINE = System.lineSeparator();
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
             + ": View all orders located at the same postal sector based on the displayed order list."
-            + NEWLINE + "Parameters: POSTAL_SECTOR (must be a valid postal sector)"
+            + NEWLINE + "Parameters: POSTAL_SECTOR or Area"
             + NEWLINE + "A postal sector is the first two digits of a six digit Singapore postal code"
-            + NEWLINE + "Example: " + COMMAND_WORD + " 14";
+            + NEWLINE + "An area is one of the following: Central, East, North-East, West, North"
+            + NEWLINE + "Example: " + COMMAND_WORD + " 14"
+            + NEWLINE + "Example: " + COMMAND_WORD + " central";
 
-    public static final String MESSAGE_SUCCESS = "Displayed all orders in postal sector."
+    public static final String MESSAGE_SUCCESS_POSTAL_SECTOR = "Displayed all orders in postal sector."
             + NEWLINE + "General Location: %1$s";
-    public static final String MESSAGE_FAILURE = "Invalid postal sector given.";
+    public static final String MESSAGE_SUCCESS_AREA = "Displayed all orders in area (%s)";
+    public static final String MESSAGE_FAILURE_POSTAL_SECTOR = "Invalid postal sector given.";
+    public static final String MESSAGE_FAILURE_AREA = "Invalid area given.";
 
-    private final Index postalSector;
+    private final String searchTerm;
+    private final List<String> validAreas = new ArrayList<>(Arrays.asList(
+            "CENTRAL",
+            "EAST",
+            "NORTH-EAST",
+            "WEST",
+            "NORTH"));
```

Comment:
Is it better to make a enum class or perhaps like the CliSyntax of the original AB3 code? Is it possible anywhere else will be using region as well?